### PR TITLE
fix: apply legend to all numeric and boolean types (DHIS2-17611)

### PR DESCRIPTION
**Pull request by janhenrikoverland**
Implements [DHIS2-17611](https://dhis2.atlassian.net/browse/DHIS2-17611)

---

### Description

This PR does two things:

- Applies legends to all numeric and boolean values, not just value type `NUMBER`, to support event data items
- Renders booleans as numeric values (integers), ie `.0` is stripped so that the number of "Yes" is presented as `3` instead of `3.0`

---

### TODO

-   [ ] Manual testing
-   [x] Figure out the removed i18n

---

### Screenshots

Before
Screenshot from 2024-06-18 15-43-06

After
Screenshot from 2024-06-18 15-43-10

(The negative integer value is just out of bounds of the legend set)

[DHIS2-17611]: https://dhis2.atlassian.net/browse/DHIS2-17611?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/components/PivotTable/PivotTableValueCell.js [hunk 1]
```diff
@@ -46,7 +49,8 @@ export const PivotTableValueCell = ({
     // TODO: Add support for 'INTEGER' type (requires server changes)
```
Comment (edoardo) [inline src/components/PivotTable/PivotTableValueCell.js]: Does this comment still applies?
`isNumericValueType` returns `true` for integer types.

File: i18n/en.pot [hunk 2]
```diff
@@ -38,11 +38,6 @@ msgstr "Created {{time}} by {{author}}"
 msgid "Created {{time}}"
 msgstr "Created {{time}}"
 
-msgid "Viewed {{count}} times"
```
Comment (edoardo) [inline i18n/en.pot]: I don't see where these strings were removed in this PR.
Was it perhaps done in another PR but the pot file was not updated?

**Review by edoardo — CHANGES_REQUESTED** [review]
There is an extra `output.tgz` file which shouldn't end up in master.

File: i18n/en.pot (hunk 2 above)
Comment (janhenrikoverland) [reply i18n/en.pot]: Looks like the issue where i18n changes are not reflected in the pot files before yarn start or build have been run. So quite often the pot file is updated in the next PR. Not great.

**Review by edoardo — APPROVED** [review]
